api/socket/socket.proto

```diff
@@ -0,0 +1,21 @@
+// Copyright 2024 TII (SSRC) and the Ghaf contributors
+// SPDX-License-Identifier: Apache-2.0
+syntax = "proto3";
+option go_package = "./socketproxy";
+package socketproxy;
+
+message Empty {}
+
+message SyncData {
+    uint32 Status = 1;
+}
+
+message SocketDataStream {
+    bytes Data = 1;
+}
+
+service SocketStream {
+    rpc Sync(SyncData) returns (SyncData) {}
+    rpc ReceiveData(stream SocketDataStream) returns (Empty) {}
```

Comment:
To me the design would seem simpler, if there was a bi-directional RPC; i.e. something like `rpc Connect(stream SocketDataStream) returns (stream SocketDataStream) {}`

Reply:
That was my first design as well, then I wanted to make it client/server side agnostic so that either side can initiate connections. Now that for the current dbus setup it isn't required, I can indeed revert that back into a true client/server model.